### PR TITLE
Releases/Versioning mechanism

### PR DESCRIPTION
**Pull request by rekram1-node**
This action will automate the tagging and releases when the master branch is updated

Note you will need to expose a github personal access token in the project secrets, this cannot be seem by people but the publish will require the action. @anthdm 

add a secret named: GITHUB_PERSONAL_ACCESS_TOKEN and give it the access token < the token should not need very many permissions

Changes:
* new github action
* goreleaser config < just tells the releaser that this is a library and there is no main package essentially

### CODE REVIEW

**Comment by anthdm** [comment]
@rekram1-node  Thanks!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.x
```
Comment (andrewpsp) [inline .github/workflows/release.yml:18]: In the spirit of Semver to ensure at least 1.19 is being used  we should be using the 
```
go-version: '>=1.19.0'

**Review by andrewpsp — CHANGES_REQUESTED** [review]
Pin golang version to last stable or higher.

File: .github/workflows/release.yml (hunk 1 above)
Comment (rekram1-node) [reply .github/workflows/release.yml]: @andrewpsp good point fixed

**Comment by anthdm** [comment]
Merging this soon boys! Thanks

**Pull request closed without merging**